pkg/api/types.go

```diff
@@ -204,6 +204,8 @@ type VolumeSource struct {
 
 // Similar to VolumeSource but meant for the administrator who creates PVs.
 // Exactly one of its members must be set.
+// All PersistentVolumes are volumes, only some Volumes are persistent.  All pointers
```

Comment:
I think we're saying the same thing.  VS and PVS are in the API and contain whatever pointers to volumes as appropriate.  There is an internal type (not in the API) that contains all the pointers.  VS and PVS translate to the internal type.  Volume plugins work with the internal type.  Is that correct?